x-pack/plugin/sql/sql-client/src/test/java/org/elasticsearch/xpack/sql/client/VersionTests.java

```diff
@@ -43,4 +53,44 @@ public void testInvalidVersion() {
         IllegalArgumentException err = expectThrows(IllegalArgumentException.class, () -> Version.from("7.1"));
         assertEquals("Invalid version 7.1", err.getMessage());
     }
+    
+    public void testVersionFromJarInJar() throws IOException {
+        Path dir = createTempDir();
+        Path jarPath = dir.resolve("foo.jar");              // simulated uberjar containing the jdbc driver
+        Path innerJarPath = dir.resolve("es-sql-jdbc.jar"); // simulated ES JDBC driver file
```

Comment:
Given it's a test, this is completely optional: extract the `"es-sql-jdbc.jar"` into a final var and reuse that below.